src/routes/transactions/mappers/queued-items/queued-items.mapper.ts

```diff
@@ -0,0 +1,137 @@
+import { Injectable } from '@nestjs/common';
+import { Safe } from '../../../../domain/safe/entities/safe.entity';
+import { ConflictType } from '../../entities/conflict-type.entity';
+import { MultisigExecutionInfo } from '../../entities/multisig-execution-info.entity';
+import { QueuedItem } from '../../entities/queued-item.entity';
+import { ConflictHeaderQueuedItem } from '../../entities/queued-items/conflict-header-queued-item.entity';
+import {
+  LabelItem,
+  LabelQueuedItem,
+} from '../../entities/queued-items/label-queued-item.entity';
+import { TransactionGroup } from '../../entities/queued-items/transaction-group.entity';
+import { TransactionQueuedItem } from '../../entities/queued-items/transaction-queued-item.entity';
+import { Transaction } from '../../entities/transaction.entity';
+import { MultisigTransactionMapper } from '../multisig-transactions/multisig-transaction.mapper';
+
+@Injectable()
+export class QueuedItemsMapper {
+  constructor(private readonly mapper: MultisigTransactionMapper) {}
+
+  async getQueuedItems(
+    transactions: Transaction[],
+    safe: Safe,
+    previousPageLastNonce: number | null,
+    nextPageFirstNonce: number | null,
+  ): Promise<QueuedItem[]> {
+    const transactionGroups = this.groupByNonce(transactions);
+    let items: QueuedItem[] = [];
+    let lastProcessedNonce = previousPageLastNonce ?? -1;
+
+    transactionGroups.forEach((transactionGroup) => {
+      const { nonce } = transactionGroup;
+      if (lastProcessedNonce < safe.nonce && nonce === safe.nonce) {
+        items.push(new LabelQueuedItem(LabelItem.Next));
+      } else if (lastProcessedNonce <= safe.nonce && nonce > safe.nonce) {
+        items.push(new LabelQueuedItem(LabelItem.Queued));
+      }
+      lastProcessedNonce = nonce;
+
+      const isEdgeGroup = nonce === nextPageFirstNonce;
+      const isSingleItemGroup = transactionGroup.transactions.length === 1;
+      const conflictFromPreviousPage = nonce === previousPageLastNonce;
+      const hasConflicts = !isSingleItemGroup || isEdgeGroup;
+      if (hasConflicts && !conflictFromPreviousPage) {
+        items.push(new ConflictHeaderQueuedItem(nonce));
+      }
+
+      items = [
+        ...items,
+        ...this.getMappedTransactionGroup(
+          hasConflicts,
+          conflictFromPreviousPage,
+          isEdgeGroup,
+          transactionGroup,
+        ),
+      ];
+    });
+
+    return items;
+  }
+
+  private getMappedTransactionGroup(
+    hasConflicts: boolean,
+    conflictFromPreviousPage: boolean,
+    isEdgeGroup: boolean,
+    transactionGroup: TransactionGroup,
+  ): TransactionQueuedItem[] {
+    const firstItem = this.getAsFirstTransaction(
+      hasConflicts,
+      conflictFromPreviousPage,
+      transactionGroup.transactions[0],
```

Comment:
Every group should have at least one transaction (because of the grouping implementation), it's related to [this comment](https://github.com/5afe/safe-client-gateway-nest/pull/215#discussion_r1062685266).